scripts/launch/demo-cluster/default.nix

```diff
@@ -0,0 +1,144 @@
+  # to force the `stop_cardano` function to exit with a different code.
+  EXIT_STATUS=0
+  source ${src + "/scripts/common-functions.sh"}
+  LOG_TEMPLATE=${src + "/log-configs/template-demo.yaml"}
+  function stop_cardano {
+    trap "" INT TERM
+    echo "Received TERM!"
+    echo "Started shutdown procedure at: $(date)"
+    echo "Stopping Cardano core nodes"
+    for pid in ''${core_pid[@]}
+    do
+      echo killing pid $pid
+      kill $pid
+    done
+    ${ifWallet ''
+      echo killing wallet pid $wallet_pid
+    kill $wallet_pid
+    ''}
+    wait
+    echo "Stopped all Cardano processes, exiting with code $EXIT_STATUS!"
+    exit $EXIT_STATUS
+  }
+  system_start=$((`date +%s` + 15))
+  echo "Using system start time "$system_start
+
+
+
+  # Remove previous state
+  rm -rf ${stateDir}
+  mkdir -p ${stateDir}
+  echo "Creating genesis keys..."
+  cardano-keygen --system-start 0 generate-keys-by-spec --genesis-out-dir ${stateDir}/genesis-keys --configuration-file ${configFiles}/configuration.yaml
+
+  echo "Generating Topology"
+  gen_kademlia_topology ${builtins.toString (numCoreNodes + 1)} ${stateDir}
+
+  trap "stop_cardano" INT TERM
+  echo "Launching a demo cluster..."
+  mkdir -p ${stateDir}/logs/state-demo/logs
+  for i in {0..${builtins.toString (numCoreNodes - 1)}}
+  do
+    node_args="$(node_cmd $i "" "$system_start" "${stateDir}" "" "${stateDir}/logs" "${stateDir}") --configuration-file ${configFiles}/configuration.yaml"
+    touch ${stateDir}/logs/state-demo/logs/node''${i}.log
+    echo Launching core node $i with args: $node_args
+    cardano-node-simple $node_args &> /dev/null &
+    core_pid[$i]=$!
+
+  done
+  ${ifWallet ''
+    #export LC_ALL=en_GB.UTF-8
+    if [ ! -d ${stateDir}/tls-files ]; then
+      mkdir -p ${stateDir}/tls-files
+      openssl req -x509 -newkey rsa:2048 -keyout ${stateDir}/tls-files/server.key -out ${stateDir}/tls-files/server.crt -days 30 -nodes -subj "/CN=localhost"
+    fi
+    echo Launching wallet node:
+    i=${builtins.toString numCoreNodes}
+    wallet_args=" --tlscert ${stateDir}/tls-files/server.crt --tlskey ${stateDir}/tls-files/server.key --tlsca ${stateDir}/tls-files/server.crt"
+    # TODO: remove wallet-debug and use TLS when the tests support it
+    wallet_args="$wallet_args --wallet-address 127.0.0.1:8090 --wallet-db-path ${stateDir}/wallet-db --wallet-debug"
+    node_args="$(node_cmd $i "$wallet_args" "$system_start" "${stateDir}" "" "${stateDir}/logs" "${stateDir}") --configuration-file ${configFiles}/configuration.yaml"
+    touch ${stateDir}/logs/state-demo/logs/node''${i}.log
+    grep "INFO" ${stateDir}/logs/state-demo/logs/node''${i}.log &
+    echo ${stateDir}/logs/state-demo/logs/node''${i}.log
+    echo Running wallet with args: $node_args
+    cardano-node $node_args &> /dev/null &
+    echo "wallet started at $(date)"
+    wallet_pid=$!
+  ''}
+  # Query node info until synced
+  SYNCED=0
+  while [[ $SYNCED != 100 ]]
```

Comment:
The meaning of `$SYNCED` is overloaded. Until `$PERC` reaches 100 it refers to the current iteration of the while loop, up to the 10th iteration.

Suggested improvements:
- Set `PERC=0` before the while loop and do `while [[ $PERC != 100 ]]`
- Rename `SYNCED` to something like `TRIES`